python/cudaq/kernel/register_op.py

```diff
@@ -58,5 +59,8 @@ def kernel():
     # Make available to kernel builder object
     setattr(PyKernel, operation_name,
             partialmethod(__generalCustomOperation, operation_name))
+    # Let the runtime know about this registered operation.
+    # Note: the matrix generator/construction is not known by the ExecutionManager in this case since we don't expect the ExecutionManager to be involved.
+    cudaq_runtime.register_custom_operation(operation_name)
```

Comment:
Q: Why do we need this?
(Is it for remote execution support?)

Reply:
The `NoiseModel` class has a validation check for gate name when adding noise channels. 
Previously, we only check against built-in gates. To support custom ops, I moved the custom gate map/registry to `CustomOp.h` in `common/` so that the `NoiseModel` can check against registered custom ops. Python has a different registration mechanism here; hence we need to propagate this to the C++ registry as well via this call.